### PR TITLE
Update Service Bus libraries from ServerCommon

### PR DESCRIPTION
**Pull request by loic-sharma**
See: https://github.com/NuGet/ServerCommon/pull/359

Tracked by: https://github.com/NuGet/Engineering/issues/3650

DEV build: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4585079&view=results
DEV test release: https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=989991

NuGetGallery.Core build: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4585250&view=results

### CODE REVIEW

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 1]
```diff
@@ -65,7 +65,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Validation">
-      <Version>2.84.0</Version>
+      <Version>2.84.0-nikhgup-serviceBusUpdate-4462793</Version>
```
Comment (loic-sharma) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj:68]: TODO

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 2]
```diff
@@ -65,7 +65,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Validation">
-      <Version>2.84.0</Version>
+      <Version>2.85.0-nikhgup-serviceBusUpdate-4466927</Version>
```
Comment (loic-sharma) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj:68]: TODO

**Review by zhhyu — APPROVED** [review]

File: src/NuGetGallery/NuGetGallery.csproj [hunk 3]
```diff
@@ -2293,7 +2293,7 @@
       <Version>1.7.0</Version>
     </PackageReference>
     <PackageReference Include="WindowsAzure.ServiceBus">
-      <Version>4.1.3</Version>
+      <Version>6.0.3</Version>
```
Comment (joelverhagen) [inline src/NuGetGallery/NuGetGallery.csproj:2296]: Do we need this explicit reference since it comes transitively?

**Review by joelverhagen — APPROVED** [review]

File: src/NuGetGallery/NuGetGallery.csproj (hunk 3 above)
Comment (loic-sharma) [reply src/NuGetGallery/NuGetGallery.csproj]: Good catch. I believe we can remove this. I will re-verify on DEV before merging this in.

**Review by lyndaidaii — APPROVED** [review]

File: tests/NuGetGallery.Facts/Services/AsynchronousDeleteAccountServiceFacts.cs [hunk 4]
```diff
@@ -192,8 +198,10 @@ public Task SendAsync(IBrokeredMessage message)
                 }
 
                 LastSentMessage = message;
-                return Task.FromResult(0);
             }
+
+            public void Close() => throw new NotImplementedException();
+            public Task CloseAsync() => throw new NotImplementedException();
```
Comment (loic-sharma) [inline tests/NuGetGallery.Facts/Services/AsynchronousDeleteAccountServiceFacts.cs:204]: These methods were added to the `ITopicClient` interface by: https://github.com/NuGet/ServerCommon/pull/360

**Comment by loic-sharma** [comment]
@zhhyu @joelverhagen @lyndaidaii Hi, I discarded your old approvals as I re-did this PR to upgrade to v2.86.0.  Please re-review when available :)

File: src/NuGetGallery/NuGetGallery.csproj (hunk 3 above)
Comment (loic-sharma) [reply src/NuGetGallery/NuGetGallery.csproj]: Verified on DEV!

**Review by joelverhagen — APPROVED** [review]

**Review by zhhyu — APPROVED** [review]

**Pull request merged**